pkg/kubelet/server/server.go

```diff
@@ -941,13 +939,19 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	longRunning := strconv.FormatBool(isLongRunningRequest(path))
 
-	servermetrics.HTTPRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
-
 	servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
 	defer servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Dec()
 
 	startTime := time.Now()
-	defer servermetrics.HTTPRequestsDuration.WithLabelValues(method, path, serverType, longRunning).Observe(servermetrics.SinceInSeconds(startTime))
+
+	handler := httplog.WithLogging(s.restfulCont, statusesNoTracePred, func(req *http.Request, status int, hijacked bool) {
```

Comment:
By moving this handler after `startTime` we change when the start time is calculated in the order of operations here. Was that intentional?

Reply:
Yeah, you are right. I have no much other idea to fill the `code` label of  metrics's `HTTPRequestsDuration`.

Reply:
@ehashman see updated.